framework/include/kernels/ConservativeAdvection.h

```diff
@@ -28,12 +28,25 @@ class ConservativeAdvectionTempl : public GenericKernel<is_ad>
 protected:
   virtual GenericReal<is_ad> computeQpResidual() override;
   virtual Real computeQpJacobian() override;
+  virtual Real computeQpOffDiagJacobian(unsigned int jvar) override;
   virtual void computeResidual() override;
   virtual void computeJacobian() override;
 
+  /// Scalar material property
+  const GenericMaterialProperty<Real, is_ad> & _scalar;
+
+  /// field variable
```

Comment:
More info please. E.g. a field variable whose gradient will be used as the advecting velocity